### PR TITLE
chore(path): release `path@1.0.0`

### LINKED ISSUE #4922: stabilize `@std/path`
part of #4600 

- [x] fully documented https://github.com/denoland/deno_std/pull/4900
- [ ] fully tested
- [ ] no ongoing breaking change discussion
- [x] 1.0.0-rc.1
- [ ] wait 1 month
- [ ] 1.0.0 (stabilization)

### PR DESCRIPTION
**Pull request by iuioiua**
To be merged July 4.

The latest API docs: https://jsr.io/@std/path@1.0.0-rc.3/doc

Closes #4922

### CODE REVIEW

**Comment by kt3k** [comment]
memo:
- There are 2 types with the same description: `GlobOptions` vs `GlobToRegExp`
- `FormatInputPathObject` doesn't seem aligned to the naming convention

File: deno.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "@std/media-types": "jsr:@std/media-types@^1.0.1",
     "@std/msgpack": "jsr:@std/msgpack@^1.0.0-rc.2",
     "@std/net": "jsr:@std/net@^0.224.4",
-    "@std/path": "jsr:@std/path@1.0.0-rc.3",
+    "@std/path": "jsr:@std/path@1.0.0",
```
Comment (iuioiua) [inline deno.json:42]: Should it be this? Does this make a difference?
```suggestion
    "@std/path": "jsr:@std/path@^1.0.0",
```

Comment (kt3k) [reply deno.json:42]: Thanks for catching. We need that to make package deduplication possible.

**Review by kt3k — APPROVED** [review]
LGTM

**Review by lucacasonato — APPROVED** [review]
LGTM - I am happy with this API

**Pull request merged**